staging/src/k8s.io/apiserver/pkg/endpoints/handlers/patch.go

```diff
@@ -275,15 +275,15 @@ type patcher struct {
 }
 
 type patchMechanism interface {
-	applyPatchToCurrentObject(currentObject runtime.Object) (runtime.Object, error)
+	applyPatchToCurrentObject(newObject, currentObject runtime.Object) (runtime.Object, error)
```

Comment:
I'm not sure I like that we do that, especially since it's only used by apply

Reply:
I don't see where you get the newObject to be the patch? is it done automatically?

Reply:
The current architecture seems kinda weird to me. So storing so much inside the patcher object. But i guess that's personal preference or that I am just not used to the style.

The newObject comes from line 403 and then going deeper down from line 551 which calls rest.DefaultUpdatedObjectInfo and this one then calls all transform funcs so:

```
// TransformFunc is a function to transform and return newObj
  | type TransformFunc func(ctx context.Context, newObj runtime.Object, oldObj runtime.Object) (transformedNewObj runtime.Object, err error)
```

But i might be wrong there.
changing this seemed better to me than looking at ways to get the runtime object from patchBytes without using gvkParser. But as a result it causes other problems, that's true. See https://github.com/kubernetes/kubernetes/pull/71583/files#diff-2b1fc19cc9dd827d6cf707c95695d145R95

Reply:
Well... I see why the parser is a data transfer object. there is a lot going on. but when there already is the newObject in the transforms I didn't see reason on why not to use it.

Reply:
I just looked at it again. My current solution is nothing I like all too much as well.
So either we get some way of creating a TypedValue from patchBytes or we change more.

The patcher code etc. looks very messy too me, correct me if i'm wrong. It seems to me that a lot of this patchBytes solution was because it was the easiest to do that way. What do you think about the patcher.go file? Should we just leave everything as it is and use pachBytes for getting a TypedValue to apply or would it be worth to change more on the patcher and clean it up (if that's even possible, I'm not sure I can do it on my own so I'm asking).

Reply:
> The patcher code etc. looks very messy too me

We all agree with that, but I don't think we want to focus on that right now. I'd go with the simplest solution (and I think we agree here), which is to add another convert method from patchbytes to the TypeConverter, and avoid changing to many things in `patcher.go`

Reply:
Okay so should I get the TypeParser thing back in I had before, exporting the gvkParser and then just use it there?
Or do you have a better suggestion for this?